### ISSUE TITLE
Telescope buffers not showing lnum correctly

### ISSUE THREAD

**Issue by shunlir** [nvim-telescope/telescope.nvim#1255]
### Description

**Telescope buffers** show buffers other than the current buffer with line number 1
Seems a regression of commit: https://github.com/nvim-telescope/telescope.nvim/commit/1c35ea319e847e07653cde5bb817daba8dd83a70 , `local line_count = vim.api.nvim_buf_line_count(entry.bufnr)` is `0` for buffers other than the current one.

### Neovim version

NVIM v0.5.0
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/gcc-11 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/runner/work/neovim/neovim/build/config -I/home/runner/work/neovim/neovim/src -I/home/runner/work/neovim/neovim/.deps/usr/include -I/usr/include -I/home/runner/work/neovim/neovim/build/src/nvim/auto -I/home/runner/work/neovim/neovim/build/include
Compiled by runner@fv-az242-526

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "
/home/runner/work/neovim/neovim/build/nvim.AppDir/usr/share/nvim"

### Operating system and version

Ubuntu 20.04

### checkhealth telescope

```markdown
health#telescope#check
========================================================================
## Checking for required plugins
  - OK: plenary installed.
  - WARNING: nvim-treesitter not found.

## Checking external dependencies
  - OK: rg: found ripgrep 12.1.1 (rev 7cb211378a)
  - OK: fd: found fd 8.2.1

## ===== Installed extensions =====

## Telescope Extension: `fzf`
  - INFO: No healthcheck provided
```

### Steps to reproduce

1. `nvim -nu minimal.lua`
2. `:e file_a`, move cursor to somewhere in the file
3. `:e file_b`, move cursor to somewhere in the file
4. `:lua require('telescope.builtin').buffers()`

### Expected behavior

**Telescope buffers** show line number of each buffer correctly

### Actual behavior

**Telescope buffers** show buffers other than the current buffer with line number 1
![image](https://user-images.githubusercontent.com/14887081/133884454-9fcdb636-2ee9-433f-b6e7-b716dd5d11cc.png)

### Minimal config

```Lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'
local function load_plugins()
  require('packer').startup {
    {
      'wbthomason/packer.nvim',
      {
        'nvim-telescope/telescope.nvim',
        requires = {
          'nvim-lua/plenary.nvim',
          { 'nvim-telescope/telescope-fzf-native.nvim', run = 'make' },
        },
      },
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. '/plugin/packer_compiled.lua',
      display = { non_interactive = true },
    },
  }
end
_G.load_config = function()
  require('telescope').setup()
  require('telescope').load_extension('fzf')
  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Telescope and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end
load_plugins()
require('packer').sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

**Comment by fdschmidt93** [comment]
I cannot reproduce the actual issue as per the video.

https://user-images.githubusercontent.com/39233597/133886043-7bf5d696-5956-4731-b63d-2b0538510bcc.mp4

I think what you observe is the effect from `vim.fn.getbufinfo` (see `:h getbufinfo`) that sets lines to 0 for buffers that have not yet opened in your current window (which we then set to 1). That's what I'm showing the video when I use `:Telescope find_files` to open a new file in a split and using the buffers picker in the original buffer.

**Comment by shunlir** [comment]
Thanks @fdschmidt93 ,
I switched to nvim nightly build (0.6), can't reproduce the issue as well.
The reported issue happens with 0.5 release build. (the repro' steps in the ticket use `:e` command to open file, actually the issue also happens with telescope openning the files)

**Comment by fdschmidt93** [comment]
This should now be conclusively fixed with #1400, hopefully.

In case it's not, please let me know and we'll reopen this.

(if it's not occurring with 0.6.0, it should be fixed in not so soon time upstream when 0.6.0 is released :) closing one way or another for tidyness)
